frontend/appflowy_flutter/lib/workspace/presentation/settings/widgets/settings_appearance_view.dart

```diff
@@ -158,3 +134,154 @@ class ThemeModeSetting extends StatelessWidget {
     }
   }
 }
+
+class ThemeFontFamilySetting extends StatefulWidget {
+  const ThemeFontFamilySetting({
+    super.key,
+    required this.currentFontFamily,
+  });
+
+  final String currentFontFamily;
+
+  @override
+  State<ThemeFontFamilySetting> createState() => _ThemeFontFamilySettingState();
+}
+
+class _ThemeFontFamilySettingState extends State<ThemeFontFamilySetting> {
+  final List<String> availableFonts = GoogleFonts.asMap().keys.toList();
+  final ValueNotifier<String> query = ValueNotifier('');
+
+  @override
+  Widget build(BuildContext context) {
+    return ThemeSettingDropDown(
+      label: LocaleKeys.settings_appearance_fontFamily_label.tr(),
+      currentValue: parseFontFamilyName(widget.currentFontFamily),
+      onClose: () {
+        query.value = '';
+      },
+      popupBuilder: (_) => ListView(
+        shrinkWrap: true,
+        children: [
+          FlowyTextField(
+            hintText: LocaleKeys.settings_appearance_fontFamily_search.tr(),
+            autoFocus: false,
+            debounceDuration: const Duration(milliseconds: 300),
+            onChanged: (value) {
+              query.value = value;
+            },
+          ),
+          const SizedBox(height: 4),
+          ValueListenableBuilder(
+            valueListenable: query,
+            builder: (context, value, child) {
+              var displayed = availableFonts;
+              if (value.isNotEmpty) {
+                displayed = availableFonts
+                    .where(
+                      (font) => font
+                          .toLowerCase()
+                          .contains(value.toLowerCase().toString()),
+                    )
+                    .sorted((a, b) => levenshtein(a, b))
+                    .toList();
+              }
+              return ListView.builder(
+                itemBuilder: (context, index) => _fontFamilyItemButton(
+                  context,
+                  GoogleFonts.getFont(displayed[index]),
+                ),
+                itemCount: displayed.length,
+                shrinkWrap: true,
+              );
+            },
+          ),
+        ],
+      ),
```

Comment:
I think using slivers here will improve the performance.
I also add a little padding on the right of textfield, so it won't be covered by the scrollbar.

![image](https://github.com/AppFlowy-IO/AppFlowy/assets/14248245/c1a46ec4-8101-447a-a167-492dcb4ddfc7)

```suggestion
      popupBuilder: (_) => CustomScrollView(
        shrinkWrap: true,
        slivers: [
          SliverPadding(
            padding: const EdgeInsets.only(right: 8),
            sliver: SliverToBoxAdapter(
              child: FlowyTextField(
                hintText: LocaleKeys.settings_appearance_fontFamily_search.tr(),
                autoFocus: false,
                debounceDuration: const Duration(milliseconds: 300),
                onChanged: (value) {
                  query.value = value;
                },
              ),
            ),
          ),
          const SliverToBoxAdapter(
            child: SizedBox(height: 4),
          ),
          ValueListenableBuilder(
            valueListenable: query,
            builder: (context, value, child) {
              var displayed = availableFonts;
              if (value.isNotEmpty) {
                displayed = availableFonts
                    .where(
                      (font) => font
                          .toLowerCase()
                          .contains(value.toLowerCase().toString()),
                    )
                    .sorted((a, b) => levenshtein(a, b))
                    .toList();
              }
              return SliverFixedExtentList.builder(
                itemBuilder: (context, index) => _fontFamilyItemButton(
                  context,
                  GoogleFonts.getFont(displayed[index]),
                ),
                itemCount: displayed.length,
                itemExtent: 32,
              );
            },
          ),
        ],
      ),
```